opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/reminder/SyncAccountAlarmListener.java

```diff
@@ -0,0 +1,33 @@
+package de.geeksfactory.opacclient.reminder;
+
+import com.commonsware.cwac.wakeful.WakefulIntentService;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+
+public class SyncAccountAlarmListener implements WakefulIntentService.AlarmListener {
+    // We could use a longer interval like 3 hours, but on API < 19, only the AlarmManager
+    // .INTERVAL_... constants make the AlarmManager use inexact alarm delivery for better
+    // battery efficiency. The next larger constant available would be AlarmManager
+    // .INTERVAL_HALF_DAY.
```

Comment:
So you suggest to set the regular interval to `AlarmManager.INTERVAL_HALF_DAY` and tell `SyncAccountService` to schedule an additional alarm one hour later if it fails?
I agree that these times would be enough for normal operation, but don't how to implement it in the best way:

If we have no internet connection (or no WiFi with the corresponding setting activated, or a library website is offline) for a longer time (say, a week), the service would run every (inexact) half day and schedule itself to run one hour later. Then, the "one hour later" runs would reschedule themselves to run one hour later and won't stop doing that when the next "every half day" run is reached. This would result in the service running ~14 times per hour at the end of the week (and probably not concentrated around one point in the hour due to the inexactness of the "every half day" alarms).

Another way to do it would be to determine the interval to the next run of the service in the service itself and let it only schedule the next run, like we did it before. The problem with this way is that if one run of the service is interrupted (due to low memory or a crash or whatever), the service probably won't run again until the next device reboot.

We could also use Android's [`JobScheduler`](https://developer.android.com/reference/android/app/job/JobScheduler.html) instead of `AlarmManager` for scheduling service runs, so it won't even run when there's no Internet connection available. The problem is it is only available on API 21 and above. There's also [`GcmNetworkManager`](https://developers.google.com/cloud-messaging/network-manager), but it only works on devices with Google Play Services.